server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -2798,16 +2798,24 @@ public GuestOS addGuestOs(final AddGuestOsCmd cmd) {
         guestOsVo.setIsUserDefined(true);
         final GuestOS guestOsPersisted = _guestOSDao.persist(guestOsVo);
 
-        if (cmd.getDetails() != null && !cmd.getDetails().isEmpty()) {
-            Map<String, String> detailsMap = cmd.getDetails();
-            for (Object key : detailsMap.keySet()) {
-                _guestOsDetailsDao.addDetail(guestOsPersisted.getId(), (String)key, detailsMap.get(key), false);
-            }
-        }
+        Map<String, String> details = cmd.getDetails();
+        Boolean forDisplay = cmd.getForDisplay();
```

Comment:
there are following columns in tables
`display_vm` in `vm_instance`
`display_volume` in `volumes`
`display_network` in `networks`
`display_nic` in `nics`
`display` in `vpc`
`display` in `firewall_rules`

can we add a new column `display` to `guest_os` table as well ?

Reply:
no, we can not. We painted ourselves in a corner and can not add to that table because it will exception out on upgrades.

Reply:
if we use a detail for display/not-display, it will bring more work on 
- update guest os between display/not-display
- list guest OSes by display or not-display

if we add a column in upgrade sql or in java, will there be exception during upgrade ?

Reply:
yes, there will be

Reply:
we need to change the older upgrade sql if we want to prevent this exception from occuring, which is a bad hack

Reply:
@DaanHoogland 
We can add the column with default value, then we do not need to update the SQLs.